### PR TITLE
DBZ-4028 Add test cases for KC REST extension

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-4028

### CODE REVIEW

File: debezium-connect-rest-extension/src/test/java/io/debezium/kcrestextension/TestHelper.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            .withNetwork(NETWORK)
+            .withCopyFileToContainer(
+                    MountableFile.forHostPath(
+                            "target/debezium-connect-rest-extension-1.8.0-SNAPSHOT.jar"),
+                    "/kafka/libs/debezium-kcd-rest-extension-1.8.0.jar")
+            .withKafka(KAFKA_CONTAINER.getNetwork(), KAFKA_HOSTNAME + ":9092")
+            .withLogConsumer(new Slf4jLogConsumer(LOGGER))
+            .dependsOn(KAFKA_CONTAINER);
+
+    public static KafkaContainer getKafkaContainer() {
```
Comment (rk3rn3r) [inline debezium-connect-rest-extension/src/test/java/io/debezium/kcrestextension/TestHelper.java:50]: Is this method used somewhere?

**Comment by rk3rn3r** [comment]
LGTM @ani-sha! Only see the minor comment I added inline.

File: debezium-connect-rest-extension/src/test/java/io/debezium/kcrestextension/TestHelper.java (hunk 1 above)
Comment (ani-sha) [reply debezium-connect-rest-extension/src/test/java/io/debezium/kcrestextension/TestHelper.java:50]: No, can be removed.

**Comment by rk3rn3r** [comment]
Could you maybe add some negative tests as well? Like Kafka Connect < 2.6.0 should return false on the `topic-creation` endpoint and Kafka Connect >= 2.8.0 should return false when topic creation is explicitly disabled.

**Comment by ani-sha** [comment]
@rk3rn3r The Confluent docker image provides tags from `3.0.0`. But we need to set up the `Kafka Container` testcontainers with version < 2.6.0 in order to test negative conditions.

**Comment by rk3rn3r** [comment]
@ani-sha we only need different versions from the Debezium Kafka Connect container.

File: debezium-connect-rest-extension/pom.xml [hunk 2]
```diff
@@ -24,6 +24,31 @@
...
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>kafka</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.debezium</groupId>
+      <artifactId>debezium-core</artifactId>
```
Comment (rk3rn3r) [inline debezium-connect-rest-extension/pom.xml:44]: What dependency/classes from `debezium-core` is used for the tests?

Comment (ani-sha) [reply debezium-connect-rest-extension/pom.xml]: I just removed that. It isn't required.

File: debezium-connect-rest-extension/src/test/java/io/debezium/kcrestextension/KafkaConnectResourceActiveIT.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        EXPOSED_PORT = TestHelper.getDebeziumContainer().getFirstMappedPort();
+    }
+
+    @AfterEach
+    public void stop() {
+        TestHelper.stopContainers();
+    }
+
+    @Test
+    public void testTopicCreationEndpoint() {
```
Comment (rk3rn3r) [inline debezium-connect-rest-extension/src/test/java/io/debezium/kcrestextension/KafkaConnectResourceActiveIT.java:44]: This class should not be called `...ActiveIT` as here's still the test case missing for Kafka >= 2.6.0 but `topic.creation.enable = false`.
Can you find a better name, for example just call this one `DebeziumResourceIT`, and the other one `DebeziumResourceNotActiveIT` or something so they have a relation to the tested classname.
And add this test to this test class with 1.7?
Also, maybe you want to make `1.7` a constant in the class so it's easier to update to a newer version? Or can we maybe derive it from Maven?

**Comment by rk3rn3r** [comment]
LGTM! Thx a lot @ani-sha and sorry for taking multiple review loops.

**Pull request merged**